frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx

```diff
@@ -404,9 +404,15 @@ describe("QueryBuilder", () => {
 
           userEvent.tab();
 
+          await waitFor(() => {
+            expect(
+              screen.getByRole("button", { name: "Save changes" }),
+            ).toBeEnabled();
+          });
+
           const mockEvent = callMockEvent(mockEventListener, "beforeunload");
-          expect(mockEvent.preventDefault).not.toHaveBeenCalled();
-          expect(mockEvent.returnValue).toBe(undefined);
+          expect(mockEvent.preventDefault).toHaveBeenCalled();
+          expect(mockEvent.returnValue).toBe(BEFORE_UNLOAD_UNSAVED_MESSAGE);
```

Comment:
This test is called "should trigger beforeunload event when leaving edited metadata" but was asserting the opposite.
It worked because React didn't have a chance to re-render before assertion, so I added relevant `waitFor`.